### PR TITLE
[Internal] refactored `databricks_mws_permission_assignment` to Go SDK

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
- refactored `databricks_mws_permission_assignment` to Go SDK
- added `TestUcAccAssignGroupToWorkspace` to flaky tests

## Tests

- [x] `make test` run locally
- [x] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: mws/resource_mws_permission_assignment.go [hunk 1]
```diff
@@ -2,131 +2,88 @@ package mws
...
 func ResourceMwsPermissionAssignment() common.Resource {
-	type entity struct {
-		WorkspaceId int64    `json:"workspace_id"`
-		PrincipalId int64    `json:"principal_id"`
-		Permissions []string `json:"permissions" tf:"slice_as_set"`
-	}
-	s := common.StructToSchema(entity{},
-		common.NoCustomize)
+	s := common.StructToSchema(iam.UpdateWorkspaceAssignments{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
```
Comment (alexott) [inline mws/resource_mws_permission_assignment.go:25]: Should we modify type to be a Set?

File: mws/resource_mws_permission_assignment.go [hunk 2]
```diff
@@ -2,131 +2,88 @@ package mws
...
+			list, err := acc.WorkspaceAssignment.ListByWorkspaceId(ctx, common.MustInt64(workspaceId))
 			if err != nil {
 				return err
 			}
-			permissions, err := list.ForPrincipal(common.MustInt64(principalId))
+			permissions, err := getPermissionsByPrincipal(*list, common.MustInt64(principalId))
 			if err != nil {
 				return err
 			}
 			return common.StructToData(permissions, s, d)
```
Comment (alexott) [inline mws/resource_mws_permission_assignment.go:75]: We need to set workspace ID and principal ID explicitly - otherwise, import won't work correctly

**Review by alexott — APPROVED** [review]

**Comment by alexott** [comment]
Hmmm, it looks like integration test is failing... Not sure if it's because of the eventual consistency or not

**Comment by nkvuong** [comment]
@alexott it is to do with the creation of a new group, which has eventual inconsistency itself. Maybe changing the integration test to a service principal would help?

**Comment by alexott** [comment]
Yes, I don't remember that we had problems with users/SPNs

**Review by mgyucht — APPROVED** [review]

**Pull request merged**